network/p2p/peerstore/peerstore_test.go

```diff
@@ -77,3 +79,192 @@ func TestPeerstore(t *testing.T) {
 	require.Equal(t, 7, len(peers))
 
 }
+
+func testPhonebookAll(t *testing.T, set []string, ph network.Phonebook) {
+	actual := ph.GetAddresses(len(set), network.PhoneBookEntryRelayRole)
+	for _, got := range actual {
+		ok := false
+		for _, known := range set {
+			if got == known {
+				ok = true
+				break
+			}
+		}
+		if !ok {
+			t.Errorf("get returned junk %#v", got)
+		}
+	}
+	for _, known := range set {
+		ok := false
+		for _, got := range actual {
+			if got == known {
+				ok = true
+				break
+			}
+		}
+		if !ok {
+			t.Errorf("get missed %#v; actual=%#v; set=%#v", known, actual, set)
+		}
+	}
+}
+
+func testPhonebookUniform(t *testing.T, set []string, ph network.Phonebook, getsize int) {
+	uniformityTestLength := 250000 / len(set)
+	expected := (uniformityTestLength * getsize) / len(set)
+	counts := make([]int, len(set))
```

Comment:
maybe make a map of strings to integers and prepopulate it with `set` as keys and zeros as values, then in your `i` loop, you can range actuals once and if the key is in your map, you can increment the map value.